Add scion-viz to SCIONLab VMs

mlegner opened the pull request:
This PR builds onto PR #96 and partially addresses issue #81 by including scion-viz in the SCIONLab VMs.
It depends on PR #101 that includes the `endhost` folder and `sciond` configuration.
This commit clones the scion-viz repository in all SCIONLab VMs and automatically enables the webserver.
It sets up port forwarding such that the website can be viewed outside of the VM.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/netsec-ethz/scion-coord/100)
<!-- Reviewable:end -->

ercanucan commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> (only minor comments)

---

Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[vagrant/README, line 71 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/100#-Ky6rZgzIEuAqkWBYJeg:-Ky6rZgzIEuAqkWBYJeh:b-lb53ts) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/563fb6e51e30a4e839486ef3ebbdcff1128b8f13/vagrant/README#L71)):*
> ```
> 
> 
> ## Displaying topology visualizations
> ```

minor: I would say something more like "Visualizing the Topology" or "Topology Visualization".

---

*[vagrant/scion.service, line 10 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/100#-Ky6qmWveeZU9D0nJcsY:-Ky6qmWveeZU9D0nJcsZ:b-uomc2p) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/563fb6e51e30a4e839486ef3ebbdcff1128b8f13/vagrant/scion.service#L10)):*
> ```
> RemainAfterExit=yes
> User=ubuntu
> WorkingDirectory=/home/ubuntu/go/src/github.com/netsec-ethz/scion/
> ```

minor: I would stay consistent regarding the trailing slashes. In the above file ('scion-viz.service') you did not use them while specifying WorkingDirectory.

---

*[vagrant/Vagrantfile, line 25 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/100#-Ky6qLXO5L0iyQpA4zL5:-Ky6qLXO5L0iyQpA4zL6:b4wtka1) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/563fb6e51e30a4e839486ef3ebbdcff1128b8f13/vagrant/Vagrantfile#L25)):*
> ```Ruby
>     cd scion-viz/python/web
>     pip3 install --user --require-hashes -r requirements.txt
>     python3 ./manage.py migrate
> ```

minor: this probably would not be necessary as scion-viz does not use any database of the Django framework?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/100#-:-Ky6s4VxmKRNMgU6rSR_:b-6pwbbl)*
<!-- Sent from Reviewable.io -->

ercanucan commented:
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/100)*
<!-- Sent from Reviewable.io -->

mlegner commented:
I addressed the review comments and rebased + squashed the PR together with the previous one.

---

Review status: 2 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[vagrant/README, line 71 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/100#-Ky6rZgzIEuAqkWBYJeg:-KyFjw_gke1i8PL0WUCD:b-896fix) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/563fb6e51e30a4e839486ef3ebbdcff1128b8f13/vagrant/README#L71)):*
<details><summary><i>Previously, ercanucan (ercanucan) wrote…</i></summary><blockquote>

minor: I would say something more like "Visualizing the Topology" or "Topology Visualization".
</blockquote></details>

Done.

---

*[vagrant/scion.service, line 10 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/100#-Ky6qmWveeZU9D0nJcsY:-KyFkNqAJw04s50YDMKS:b-896fix) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/563fb6e51e30a4e839486ef3ebbdcff1128b8f13/vagrant/scion.service#L10)):*
<details><summary><i>Previously, ercanucan (ercanucan) wrote…</i></summary><blockquote>

minor: I would stay consistent regarding the trailing slashes. In the above file ('scion-viz.service') you did not use them while specifying WorkingDirectory.
</blockquote></details>

Done.

---

*[vagrant/Vagrantfile, line 25 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/100#-Ky6qLXO5L0iyQpA4zL5:-KyFkcz4UFJpW92vKLzL:bdgda0k) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/563fb6e51e30a4e839486ef3ebbdcff1128b8f13/vagrant/Vagrantfile#L25)):*
<details><summary><i>Previously, ercanucan (ercanucan) wrote…</i></summary><blockquote>

minor: this probably would not be necessary as scion-viz does not use any database of the Django framework?
</blockquote></details>

I took this from the README of scion-viz. I would probably leave it in as it probably doesn't hurt and may be useful if the system changes at some point.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/100#-:-KyFkpo6yTeqlIDSGSAs:b-bsjf5l)*
<!-- Sent from Reviewable.io -->

ercanucan commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 2 of 4 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/100#-:-KyFnIWpsnbpu3REYMBe:bnfp4nl)*
<!-- Sent from Reviewable.io -->

ercanucan commented:
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/100)*
<!-- Sent from Reviewable.io -->

mwfarb commented:
:+1:  Also, first time users have asked me if the warning they get calling 'runserver' without 'migrate' first is a bug, and calling migrate helps eliminate confusion.
